### PR TITLE
Enabling client-side compression for Pub/Sub

### PR DESCRIPTION
**Pull request by davidtorres**
Partially addressing #1639, changes will need to be made to the gax.grpc classes to support compression.

### CODE REVIEW

**Comment by davidtorres** [comment]
@pongad @garrettjonesgoogle for your review

File: google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Publisher.java [hunk 1]
```diff
@@ -696,6 +705,15 @@ public Builder setExecutorProvider(ExecutorProvider executorProvider) {
       this.executorProvider = Preconditions.checkNotNull(executorProvider);
       return this;
     }
+    
+    /** 
+     * Gives the ability to disable client-side compression. 
+     * Note compression is enabled by default. 
+     */
+    public Builder setCompressionEnabled(boolean enabled) {
```
Comment (pongad) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Publisher.java]: The name `setCompressionEnabled` sounds a little strange to me. That said there are java UI compoenents with method `setEnabled` so I think the name is not that jarring.

Comment (garrettjonesgoogle) [inline google-cloud-pubsub/src/main/java/com/google/cloud/pubsub/spi/v1/Publisher.java]: I think this is the established pattern.

**Comment by garrettjonesgoogle** [comment]
LGTM too.

**Comment by tomerr-ma** [comment]
Hi, Is there any plan to evaluate compression/gzip support?

**Comment by tomarikawa** [comment]
I could not quite get, why is this reverted? Could somebody explain, please.

**Pull request merged**